sklearn/decomposition/pca.py

```diff
@@ -175,25 +175,30 @@ class PCA(_BasePCA):
             0 < n_components < min(X.shape)
         randomized :
             run randomized SVD by the method of Halko et al.
+        lobpcg :
+            run Locally Optimal Block Preconditioned Conjugate Gradient
+            (LOBPCG) by Knyazev 2001 for a normal matrix of the smallest size.
```

Comment:
I don't know what "for a normal matrix of the smallest size" is in reference to. Is this what lobpcg takes as input or produces as output? What does "normal" mean? What does "the smallest size" mean?

Reply:
Fixed in https://github.com/scikit-learn/scikit-learn/pull/12319/commits/34b959840fa9e9a17b1cbba968ff015f8d5c70fa 
`a normal matrix X'*X or X*X',
            whichever of the two is of the smallest size`
Thanks!